daemon/cmd/daemon_main.go

```diff
@@ -1758,11 +1759,30 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 			// This must wait for both K8s watcher caches to be synced and local endpoint restoration to be complete.
 			// Note: Synchronization of endpoints to their CEPs may not be complete at this point, but we only have to
 			// know what endpoints exist post-restoration in our endpointManager cache to perform cleanup.
-			if err := d.cleanStaleCEPs(context.Background(), d.endpointManager, params.Clientset.CiliumV2(), option.Config.EnableCiliumEndpointSlice); err != nil {
-				log.WithError(err).Error("Failed to clean up stale CEPs")
+			bo := wait.Backoff{
+				Duration: 500 * time.Millisecond,
+				Factor:   1,
+				Jitter:   0.1,
+				Steps:    5,
+				Cap:      0,
+			}
+			err := wait.ExponentialBackoffWithContext(context.Background(), bo, func(ctx context.Context) (done bool, err error) {
```

Comment:
One thing I'm wondering here is how we decide between using this approach vs pkg/controller (i.e. having a controller that runs until its error returns nil.

Also, another thought is that perhaps this might be a good case for the new "Job" type which supports retry until completion for OneShot jobs: https://github.com/cilium/cilium/blob/main/pkg/hive/job/job.go#L204

Reply:
One part of the criteria is the older stable branches won't have "Job" type.

Reply:
Controller pattern is the typical pattern in the Cilium code base. It already has logic to retry exponentially. It seems like it integrates well here in this case, so we might as well utilize it. 

With the controller pattern, it becomes easier to hold a queue of items that need to be processed and go through them one by one. If one fails, that item gets put back onto the queue. The controller will exit when the queue is empty. In this case, the item in the queue here is a stale CEP.

Reply:
> One part of the criteria is that older stable branches won't have "Job" type.

Yup, good point 👍

Reply:
agreed, controller seems like the right fit here

Reply:
Somehow using a controller for this use case feels a bit heavy, the current impl already handles exp retries + retry limit. Possible with a controller of course but does not seems like the right use case for it in this instance imho.

Reply:
While the difference between a controller and the current state of the PR is minimal in terms of an actual mechanism for retries, controllers have a bit more structure around them. Controllers have a status and that is aggregated and reported to (1) metrics and (2) `cilium status` output. If this controller is failing, the user could easily be informed by the controller infrastructure, while the current PR today would have to add code to come to parity. In this case, there's no added benefit to duplicating the code for the PR to come to parity with what `pkg/controller` already provides. Does this make sense?

Reply:
After discussing this with @christarazi, the current implementation is a better fit for this particular use case